xdsl/parser/core.py

```diff
@@ -736,17 +736,19 @@ def parse_operation(self) -> Operation:
 
         return op
 
-    def _get_op_by_name(self, name: str) -> type[Operation]:
+    def _get_op_by_name(
+        self, name: str, *, is_generic: bool = False
+    ) -> type[Operation]:
```

Comment:
I am not sure I understand what is happening here; `is_generic` is not the greatest name and the way this is described, I'd expect the parser to tell us if the op is in generic or custom format and then decide based on that (along with the cmdline option for unregistered dialects)?